### PR TITLE
Drop method for KeyTable

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: python/hail/keytable.py [hunk 1]
```diff
@@ -577,6 +577,29 @@ def select(self, column_names):
...
+
+        Assume ``kt1`` is a :py:class:`.KeyTable` with three columns: C1, C2 and
+        C3.
+
+        Drop columns:
+
+        >>> kt_result = kt1.drop(['C1', 'C2'])
+
+        :param column_names: List of columns to be dropped.
+        :type: list of str
```
Comment (tpoterba) [inline python/hail/keytable.py]: can we accept str or list of str?

**Comment by jigold** [comment]
@tpoterba back to you.

File: python/hail/keytable.py [hunk 2]
```diff
@@ -567,16 +567,54 @@ def select(self, column_names):
...
+        elif not isinstance(column_names, str) and not isinstance(column_names, unicode):
+            raise TypeError("expected str or list of str for parameter 'column_names', but found %s" % type(column_names))
+
+        column_names = wrap_to_list(column_names)
         new_key = [k for k in self.key if k in column_names]
         return KeyTable(self.hc, self._jkt.select(column_names, new_key))
 
     @handle_py4j
+    @typecheck_method(column_names=oneof(strlike, listof(strlike)))
+    def drop(self, column_names):
```
Comment (tpoterba) [inline python/hail/keytable.py]: Awesome. I think that actually this should take varargs, but that's true of a bunch of methods and it's better to be consistent with the other things in the 0.1 interface.

Comment (jigold) [inline python/hail/keytable.py]: I agree.

File: python/hail/keytable.py [hunk 3]
```diff
@@ -567,16 +567,54 @@ def select(self, column_names):
...
+
+        >>> kt_result = kt1.drop(['C1', 'C2'])
+
+        :param column_names: List of columns to be dropped.
+        :type: str or list of str
+
+        :return: Key table with dropped columns.
+        :rtype: :class:`.KeyTable`
+        """
+        if isinstance(column_names, list):
```
Comment (tpoterba) [inline python/hail/keytable.py]: delete this type check stuff. It's done already

**Pull request merged**